statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,538 @@
+            Object of class SurveyMean, SurveyTotal, SurveyPercentile, etc
+
+        Returns
+        -------
+        est : ndarray
+            The point estimates of the statistic, calculated on the columns
+            of data.
+        vc : square ndarray
+            The variance-covariance matrix of the estimates, obtained using
+            the (drop 1) jackknife procedure.
+        pseudo : ndarray
+            The jackknife pseudo-values.
+        """
+
+        jdata = []
+        # for each cluster
+        for c in range(self.design.nclust):
+            # get stratum that the cluster belongs in
+            s = self.design.sfclust[c]
+            nh = self.design.ncs[s]
+            self.w = self.design.weights.copy()
+            # all weights within the stratum are modified
+            self.w[self.design.strat == s] *= nh / float(nh - 1)
+            # but if you're within the cluster to be removed, set as 0
+            self.w[self.design.sclust == c] = 0
+            # if dealing w/ survey quantile...
+            if index is not None:
+                # 3d array, nclust x col x len(quantiles)
+                jdata.append(self._stat(self.w, index))
+            else:
+                jdata.append(self._stat(self.w))
+        jdata = np.asarray(jdata)
+        nh = self.design.ncs[self.design.sfclust].astype(np.float64)
+        # pseudo = jdata + nh[:, None] * (np.dot(self.w, stat.data) - jdata)
+
+        if index is not None:
+            est = self._stat(self.design.weights, index)
+        else:
+            est = self._stat(self.design.weights)
+
+        if self.mse:
+            print('mse method')
+            jdata -= est
+        else:
+            for s in range(self.design.nstrat):
+                # get indices of all clusters within a stratum
+                ii = np.flatnonzero(self.design.sfclust == s)
+                # center the 'delete 1' statistic
+                jdata[ii, :] -= jdata[ii, :].mean(0)
+        u = np.sqrt((nh - 1) / nh)
+        jdata = u[:, None] * jdata
+        vc = np.dot(jdata.T, jdata)
+
+
+        # if not working w/ percentile and est is []
+
+        return est, np.sqrt(np.diag(vc))
+
+
+class SurveyMean(SurveyStat):
+    """
+    Calculates the mean for each column.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+    data : ndarray
+        nxp array of the data to calculate the mean on
+    method: string
+        User inputs whether to get bootstrap or jackknife SE
+
+    Attributes
+    ----------
+    data : ndarray
+        The data which to calculate the mean on
+    design :
+        Points to the SurveyDesign object
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vc : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+    def __init__(self, design, data, se_method, mse=False, replicates=None):
```

Comment:
We might want to consider a design in which instead of passing in `se_method`, the user is responsible for calling either the `jack()` or `boot()` methods (or othe rmethods to be added later).  This is like the `fit()` method we have in most of the models.  The results are only available after calling `jack` or `boot`.